web/src/app/services/selected-term.service.ts

```diff
@@ -0,0 +1,109 @@
+import {Injectable} from '@angular/core';
+import {Term} from 'yacs-api-client';
+import {Subject, Subscription} from 'rxjs';
+
+@Injectable()
+export class SelectedTermService {
+  // a subject that can be written to and subscribed to for the selected term
+  public selectedTerm: Subject<Term>;
+  // hold the terms in two forms: a mapping between the uuid and the term, and an array for ordering
+  protected terms: Map<string, Term>;
+  // cache the current ordinal and uuid
+  protected _currentOrdinal: number;
```

Comment:
why are we using underscores here but not elsewhere? We don't use this convention elsewhere in the app either

Reply:
addressed